src/main/java/bio/terra/common/FutureUtils.java

```diff
@@ -0,0 +1,79 @@
+package bio.terra.common;
+
+import bio.terra.app.controller.exception.ApiException;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public final class FutureUtils {
+
+    private FutureUtils() {
+    }
+
+    public static <T> List<T> waitFor(final List<Future<T>> futures) {
+        return waitFor(futures, Optional.empty());
+    }
+
+    /**
+     * Wait for a list of {@link Future} objects to complete and returns a list of the resolved values.
+     * If any of the threads fail, throws the first found instance of failure (non-deterministic) but waits for all
+     * futures to resolve.
+     * @param futures The {@link List} of futures to wait for
+     * @param maxThreadWait The maximum of time to wait per thread
+     * @param <T> The type that should be returned for the list of futures
+     * @return The resolved values of the futures.  There is no guarantee that the order of returned values matches the
+     * order of the passed in futures
+     */
+    public static <T> List<T> waitFor(final List<Future<T>> futures,
+                                      final Optional<Duration> maxThreadWait) {
```

Comment:
What is the benefit of using `Optional<>` over the object itself in this case? As I understand it, the intended use of `Optional` is as a function return value. For an argument, using `Duration` and allowing `null` provides the same feature.